doc/introduction/compiling_circuits.rst

```diff
@@ -332,3 +338,69 @@ observables and coefficients:
 [[0.97, 4.21], [1.43]]
 
 This and more logic to manipulate Pauli observables is found in the :mod:`~.pennylane.grouping` module.
+
+
+Just-in-time compilation with Catlyst
+-------------------------------------
+
+In addition to quantum circuit transformations, PennyLane also supports full
+hybrid just-in-time (JIT) compilation via `Catalyst
+<https://github.com/pennylaneai/catalyst>`__. Catalyst allows you to compile
+the entire quantum-classical workflow, including any optimization loops,
+which allows for optimized performance, and the ability to run the entire
+workflow on accelerator devices as appropriate.
+
+Currently, Catalyst must be installed separately, and only supports the JAX
+interface and ``lightning.qubit``. Check out the Catalyst documentation for
+`installation instructions
+<https://docs.pennylane.ai/projects/catalyst/en/latest/dev/installation.html>`__.
+
+Using Catalyst with PennyLane is a simple as using the ``@qjit`` decorator to
+compile your hybrid workflows:
+
+.. code-block:: python
+
+    from catalyst import qjit
+    from jax import numpy as jnp
+
+    dev = qml.device("lightning.qubit", wires=2, shots=1000)
+
+    @qml.qnode(dev)
+    def cost(params):
+        qml.Hadamard(0)
+        qml.RX(jnp.sin(params[0]) ** 2, wires=1)
+        qml.CRY(params[0], wires=[0, 1])
+        qml.RX(jnp.sqrt(params[1]), wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+The ``qjit`` decorator can also be used on hybrid cost functions -- that is,
+cost functions that include both QNodes and classical processing. We can even
+JIT compile the full optimization loop, for example when training models:
+
+.. code-block:: python
+
+    import jaxopt
+
+    @qjit
+    def optimization():
+        # initial parameter
+        params = jnp.array([0.54, 0.3154])
+
+        # define the optimizer
+        opt = jaxopt.GradientDescent(cost, stepsize=0.4)
```

Comment:
@dime10 my example here is assuming the JAX integration, which actually really simplifies this example -- you don't need to wrap the cost function to manually compute the gradient!

Reply:
Does this work given that it's inside the qjit decorator?

I think it would only work if we had a qnode primitive inside the JAXPR.

Reply:
@dime10 you're right, I got ahead of myself

Reply:
I think it would work if we did

```python
@jax.jit
@qjit
@qnode(dev)
```
lol

Reply:
The `jax.jit` on the qnode is not necessary, but I suppose you could do :
```py
@qjit
@qml.qnode
def cost():
   ...


@jax.jit
def workflow():
   ...
   jaxopt.GradientDescent(cost, stepsize=0.4)
```

Reply:
Yep, this is exactly what I was going for 🤦 Thanks @dime10